esx_service/utils/vm_listener.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python
+# Copyright 2016 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''
+VM change listener (started as a part of vmdkops service).
+It monitors VM poweroff events and detaches the DVS managed
+volumes from the VM and updates the status in KV
+'''
+
+import logging
+import os
+import os.path
+import atexit
+import time
+import http.client
+
+import threadutils
+import log_config
+import vmdk_utils
+import vmdk_ops
+
+from pyVmomi import VmomiSupport, vim, vmodl
+# vim api version used - version11
+
+VM_POWERSTATE = 'runtime.powerState'
+POWERSTATE_POWEROFF = 'poweredOff'
+HOSTD_RECONNECT_INTERVAL = 5 #approx time for hostd to comeup is 10-15 seconds
```

Comment:
I would start with 2secs sleep and increase HOSTD_RECONNECT_ATTEMPT if absolutely needed so that we do not miss any poweroff events.